### ISSUE TITLE
Any support for the tyring V2

### ISSUE THREAD

**Issue by ozomax1997** [FxPape/fdclct#3]
Keyboard also uses python would be great to see some development for it

**Comment by LukePrior** [comment]
Would love to see this too, i would try but no idea where to start. But if anyone can guide me in the right direction i'll have a shot at it.

**Comment by userx14** [comment]
I would guess that Drevo didn't reinvent the weel and used the same .DLL to controll the led's. Is there a Windows utility to controll the lights? And what do you want to achive? An alternative Python controll utility or something else?

**Comment by LukePrior** [comment]
There is a windows utilitity, I just want to see if I can tie the lights into system specifications like temperature etc

**Comment by userx14** [comment]
Hmmm, the structure of the tryfing app seems quite different from that of the calibur keyboard. I would guess they handle the communication with the file "yourinstallDIR/app/wkb.kb_hidapi.pyd" which unfortunately is a compiled file.
Found the following inside the wkbapp.config file:
`"APP_HID_VID": "0x0416",
    "APP_HID_PID": "0xA0F8"`

**Comment by Fail-0815** [comment]
Well, i don't have the Tyring Keyboard, but my reverse engineering process was as follows:
1. Get Windows running in a VM in some Linux System
2. Pass the USB bus on which the keyboard is connected to the VM
3. Hook Wireshark to sniff USB Packets
4. Run the Windows application and record what it does.

I'd guess they still use a very similar protocoll compared to the Calibur. You could try to set the USB-Product and/or Vendor ID in usblinux.py/usbwindows.py and just try to run it. It most probably will not light the correct keys, but if it changes anything they are probably very much alike on the Keyboard side of thing.

**Comment by LukePrior** [comment]
Thanks I shall have to give it a go sometime.

**Comment by ozomax1997** [comment]
I may be able to give it a go 🤷🏻‍♂️ 

I'm not very knowledgeable in this sort of thing but I'm willing to help out

**Comment by brainrom** [comment]
Tried it with another keyboard, but with the same software and VID, PID. It uses endpoint 0x82 as in, and 0x03 as out (AFAIK). But the only result I get is "Operation timed out".
UPD: Uses absolutely different protocol and other headers etc. I reversed it and will publish my fdclct fork in time around a week.

**Comment by brainrom** [comment]
https://github.com/brainrom/fdclct

**Comment by LukePrior** [comment]
What keyboard is that fork for?

Is it the tyring V2?

**Comment by brainrom** [comment]
No, but for the keyboard on the same chip and with the same vendor software. Anyway, it should be more compatible, than original version.
